### ISSUE TITLE
Unnamed instances crash `--optimize`

### ISSUE THREAD

**Issue by jkrenge** [autofitcloud/isitfit#5]
Running `isitfit` works perfectly:

```
$ isitfit
...
No cloudwatch data for: ...

Cost-Weighted Average Utilization (CWAU) of the AWS EC2 account:

Field                    Value
-----------------------  ----------
Start date               2019-06-28
End date                 2019-09-26
...
```

Then running `--optimize` fails:

```
$ isitfit --optimize
Initializing...
Fetching history...
Found a total of 65 EC2 instances
Intermediate results will be streamed to /var/folders/pd/40tqjt_56_g_2_5nv_rhjpkm0000gn/T/isitfit-intermediate-02zc6l1s.csv
Cloudtrail page 1: 1it [00:00,  5.15it/s]
Cloudtrail page 1: 1it [00:00, 12.91it/s]
First pass through EC2 instances: 66it [00:00, 146.23it/s]
Second pass through EC2 instances:  62%|█████████████████████████████████████████████████████████████████████████████████▏                                                  | 40/65 [00:08<00:08,  2.78it/s]
Generated by isitfit version 0.5.5
For more info about isitfit, check https://isitfit.autofitcloud.com ⛅
Also, consider following the Global Climate Strike news at https://twitter.com/hashtag/ClimateStrike 🌎
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/bin/isitfit", line 10, in <module>
    sys.exit(cli())
  File "/Users/julian/Library/Python/3.7/lib/python/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/Users/julian/Library/Python/3.7/lib/python/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/Users/julian/Library/Python/3.7/lib/python/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/julian/Library/Python/3.7/lib/python/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/isitfit/cli.py", line 72, in cli
    mm.get_ifi()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/isitfit/mainManager.py", line 126, in get_ifi
    l(ec2_obj, ec2_df, self, ddg_df)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/isitfit/optimizerListener.py", line 139, in per_ec2
    ec2_name = [x for x in ec2_obj.tags if x['Key']=='Name']
TypeError: 'NoneType' object is not iterable
```

Looking at `ec2_name = [x for x in ec2_obj.tags if x['Key']=='Name']` I thought we might need names. So I found 2 unnamed instances, named them and then it worked. :)

**Comment by shadiakiki1986** [comment]
Awesome. Will fix this and make a new release. Thanks for catching it!

**Comment by jkrenge** [comment]
Love this tool, really cool!

**Comment by shadiakiki1986** [comment]
Glad you like it :) This issue is fixed in version 0.6.2. You can upgrade with `pip3 install --upgrade isitfit`

**Comment by shadiakiki1986** [comment]
Hello @jkrenge   It's been a few months since you first tried out isitfit. Since then, I've improved it a lot and it's at version 0.20 now. I'll be happy to get your feedback on the latest version. You could upgrade with `pip3 install --upgrade isitfit` and then run `isitfit cost analyze` to calculate the efficiency of your AWS account, followed by `isitfit cost optimize` to get recommendations on which EC2/RDS sizes better be changed. Have a nice day!
